tests/cli/test_train_model.py

```diff
@@ -113,6 +115,49 @@ def test_command_line_override(monkeypatch, tmp_path, overrides):
         assert restart_options["architecture"]["training"]["batch_size"] == 3
 
 
+def test_train_from_options_restart_yaml(monkeypatch, tmp_path):
+    monkeypatch.chdir(tmp_path)
+    shutil.copy(DATASET_PATH_QM9, "qm9_reduced_100.xyz")
+
+    # run training with original options
+    options = OmegaConf.load(OPTIONS_PATH)
+    train_model(options)
+
+    # run training with options_restart.yaml
+    options_restart = OmegaConf.load("options_restart.yaml")
+    train_model(options_restart)
+
+
+def test_train_unknonw_arch_options(monkeypatch, tmp_path):
+    monkeypatch.chdir(tmp_path)
+    shutil.copy(DATASET_PATH_QM9, "qm9_reduced_100.xyz")
+
+    options_str = """
+    architecture:
+        name: experimental.soap_bpnn
+        training:
+            batch_size: 2
+            num_epoch: 1
+
+    training_set:
+        systems:
+            read_from: qm9_reduced_100.xyz
+            length_unit: angstrom
+        targets:
+            energy:
+            key: U0
+            unit: eV
+
+    test_set: 0.5
+    validation_set: 0.1
+    """
+    options = OmegaConf.create(options_str)
+
+    match = r"Additional properties are not allowed \('num_epoch' was unexpected\)"
```

Comment:
any chance of getting suggestions for closely matching allowed names?

Reply:
Good question and idea. I think [jsonschema](https://pypi.org/project/jsonschema/) can't do this. But I can keep this in mind and we can add this later if you agree.

Reply:
We can leave this for later yes =)

Reply:
Will be addressed in #271 !